Lib/textwrap.py

```diff
@@ -14,6 +14,40 @@
 # some Unicode spaces (like \u00a0) are non-breaking whitespaces.
 _whitespace = '\t\n\x0b\x0c\r '
 
+try:
+    from unicodedata import east_asian_width
+
+    def _len(text):
+        """Return the display length of the text in columns, according to
+        unicodedata.east_asian_width only.
+        """
+        return sum(2 if east_asian_width(char) in {'F', 'W'} else 1
+                 for char in text)
+
+    def _slice(text, index):
+        """Return the two slices of text cut to index.
+        """
+        width = 0
+        pos = 0
+        for char in text:
+            width += 2 if east_asian_width(char) in {'F', 'W'} else 1
```

Comment:
Why inlining _len(), I don't have seen performance issues and it's less readable (less pythonic)

Reply:
How do you know where to break once you have the whole value?

Reply:
width = width + cjk_wide(char) + 1

Reply:
Hello I was reading too fast. In my version there's the ``_wide`` boolean function.
So here ``width += _wide(char) + 1``

Reply:
And ``_len`` is just ``return sum(2 if _wide(char) else 1 for char in text)`` with no performance issues

Reply:
More pythonic, DRY.

Reply:
I won't bet on the performances, calling _len from _slice adds two functions calls per character (one to _len and one to sum). In one case I'm doing it on a character, and in the other case in a whole string. Yes I could also factorize this ternary to a third function, but I don't find it more readable.